ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -387,158 +393,155 @@ instance ( ShelleyCompatible proto era
       lst = ledgerState ext
       hst = headerState ext
       st  = shelleyLedgerState lst
-      ext = dlvCurrent dlv
-
-  getQueryKeySets = \case
-    GetUTxOByTxIn ks -> LedgerTables $ KeysMK ks
-    _other           -> emptyLedgerTables
-
-  tableTraversingQuery = \case
-      GetUTxOByAddress addrs ->
-        Just (TraversingQueryHandler
-              (\st -> SL.getFilteredUTxO (shelleyLedgerState $ ledgerState st) addrs)
-              emptyUtxo
-              combUtxo
-              id)
-      GetUTxOWhole           ->
-        Just (TraversingQueryHandler
-              (SL.getUTxO . shelleyLedgerState . ledgerState)
-              emptyUtxo
-              combUtxo
-              id)
-      _other -> Nothing
-    where
-      emptyUtxo                        = SL.UTxO Map.empty
-      combUtxo (SL.UTxO l) (SL.UTxO r) = SL.UTxO $ Map.union l r
 
-instance SameDepIndex (BlockQuery (ShelleyBlock proto era)) where
-  sameDepIndex GetLedgerTip GetLedgerTip
+  answerBlockQueryOne cfg qry dlv = case qry of
+    GetUTxOByTxIn ks -> do
+      values <- dlvRead dlv $ LedgerTables $ KeysMK ks
+      pure
+        . flip SL.getUTxOSubset ks
+        . shelleyLedgerState
+        . ledgerState
+        . stowLedgerTables
+        $ dlvCurrent dlv `withLedgerTables` values
+    GetCBOR qry'     ->
+      mkSerialised (encodeShelleyResult maxBound qry') <$>
+            answerBlockQueryOne cfg qry' dlv
+
+  answerBlockQueryAll cfg qry dlv = case qry of
+    GetUTxOByAddress ks -> undefined ks
+    GetUTxOWhole        -> undefined
```

Comment:
These two will never be called because we never instantiate the node with `ShelleyBlock` but instead with `HardForkBlock [ShelleyBlock]`, but still it doesn't look good to have this here

Reply:
I would still fill these in

Reply:
Done